src/nfa/nfa.cc

```diff
@@ -223,95 +221,92 @@ StateSet Nfa::get_terminating_states() const
 }
 
 //TODO: probably can be removed, trim_inplace is faster.
-void Nfa::trim_reverting(StateToStateMap* state_map)
+void Nfa::trim_reverting(StateRenaming* state_renaming)
 {
-    if (!state_map) {
-        StateToStateMap tmp_state_map{};
-        *this = get_trimmed_automaton(&tmp_state_map);
+    if (!state_renaming) {
```

Comment:
here we already use the "state_renaming", so we could use this style consitently also above?

Reply:
I would be OK with that.